sdk/component/store/local/store.go

```diff
@@ -110,59 +112,85 @@ func (m *manager) Read(ctx context.Context, instanceID uuid.UUID) ([]*ocsf.Vulne
 
 	defer stmt.Close()
 
-	var jsonFindingsStr string
-	err = stmt.
-		QueryRowContext(
+	rows, err := stmt.
+		QueryContext(
 			ctx,
 			sql.Named(LocalStoreColumnNameInstanceId.String(), instanceID.String()),
-		).
-		Scan(&jsonFindingsStr)
+		)
 	if err != nil {
 		if errors.Is(err, sql.ErrNoRows) {
 			return nil, errors.Errorf("%s: %w", instanceID.String(), store.ErrNoFindingsFound)
 		}
 		return nil, errors.Errorf("could not select findings: %w", err)
 	}
 
-	var jsonFindings []json.RawMessage
-	if err := json.Unmarshal([]byte(jsonFindingsStr), &jsonFindings); err != nil {
-		return nil, errors.Errorf("could not unmarshal json findings to []json.RawMessage: %w", err)
-	}
+	var findings []*vf.VulnerabilityFinding
+	for rows.Next() {
+		var (
+			findingID          uint64
+			jsonFindingDetails string
+		)
+
+		if err := rows.Scan(&findingID, &jsonFindingDetails); err != nil {
+			return nil, errors.Errorf("could not scan row: %w", err)
+		}
 
-	var findings []*ocsf.VulnerabilityFinding
-	for _, jsonFinding := range jsonFindings {
 		var finding ocsf.VulnerabilityFinding
-		if err := protojson.Unmarshal(jsonFinding, &finding); err != nil {
-			return nil, errors.Errorf("failed to unmarshal JSON findings to *ocsf.VulnerabilityFinding: %w", err)
+		if err := protojson.Unmarshal([]byte(jsonFindingDetails), &finding); err != nil {
+			return nil, errors.Errorf(
+				"failed to unmarshal JSON findings to *ocsf.VulnerabilityFinding for id %d: %w",
+				findingID,
+				err,
+			)
 		}
-		findings = append(findings, &finding)
+
+		findings = append(findings, &vf.VulnerabilityFinding{
+			ID:      findingID,
+			Finding: &finding,
+		})
+	}
+
+	if err := rows.Err(); err != nil {
+		return nil, errors.Errorf("could not scan rows: %w", err)
+	}
+
+	if len(findings) == 0 {
+		return nil, store.ErrNoFindingsFound
 	}
 
 	return findings, nil
 }
 
 // Write writes new vulnerabilities in JSON format in the database.
 func (m *manager) Write(ctx context.Context, instanceID uuid.UUID, findings []*ocsf.VulnerabilityFinding) error {
-	jsonFindings, err := m.marshalFindings(findings)
-	if err != nil {
-		return err
+	var (
+		placeHolders strings.Builder
+		values       []any
```

Comment:
why are the values a list of any instead of a list of strings?

Reply:
`Exec` needs a slice of `[]any` - this is changing in the next PR regardless and it's more type safe